app/org/sagebionetworks/bridge/models/schedules/IntervalActivityScheduler.java

```diff
@@ -21,14 +22,17 @@
     public List<ScheduledActivity> getScheduledActivities(SchedulePlan plan, ScheduleContext context) {
         List<ScheduledActivity> scheduledActivities = Lists.newArrayList();
         DateTime datetime = getScheduledTimeBasedOnEvent(context);
+
         if (datetime != null) {
             while(shouldContinueScheduling(context, datetime, scheduledActivities)) {
-                addScheduledActivityForAllTimes(scheduledActivities, plan, context, datetime);
+                LocalTime localTime = addScheduledActivityForAllTimes(scheduledActivities, plan, context, datetime.toLocalDate());
                 // A one-time activity with no interval (for example); don't loop
                 if (schedule.getInterval() == null) {
                     return trimScheduledActivities(scheduledActivities);
                 }
-                datetime = datetime.plus(schedule.getInterval());
+                // We reset the time portion of the timestamp to the last time that was used to schedule.
+                // Otherwise to-the-minute resolution of the start/end time boundaries won't be correct.
+                datetime = datetime.withTime(localTime).plus(schedule.getInterval());
```

Comment:
Yes, there are specific tests that fail if you don't do this (the ScheduledActivityServiceRecurringTest returns the wrong number of tasks). I can show why it occurs though I'm not sure why it was masked until these other changes were made.